File: compiler/src/main/java/io/github/qihuan92/activitystarter/compiler/entity/ClassType.java
package io.github.qihuan92.activitystarter.compiler.entity;

import com.squareup.javapoet.TypeName;

import io.github.qihuan92.activitystarter.compiler.utils.TypeUtils;

/**
 * ClassType
 *
 * @author qi
 * @since 2021/8/4
 */
public class ClassType {
    private final String className;

    public ClassType(String className) {
        this.className = className;
    }

    public TypeName java() {
        return TypeUtils.getJavaTypeName(className);
    }
}

File: compiler-ksp/src/main/java/io/github/qihuan92/activitystarter/compiler/ksp/utils/ClassType.kt
package io.github.qihuan92.activitystarter.compiler.ksp.utils

import com.squareup.kotlinpoet.ClassName

/**
 * ClassType
 *
 * @author qi
 * @since 2021/8/4
 */
class ClassType(val className: String) {
//    val javaTypeName
//        get() = className.javaTypeName
//
//    val kotlinTypeName
//        get() = className.type.asKotlinTypeName()
//
//    val javaClassName: com.squareup.javapoet.ClassName
//        get() = com.squareup.javapoet.ClassName.bestGuess(className)
//
    val kotlinClassName
        get() = ClassName.bestGuess(className)
}